src/modules/site-v2/templates/tools/phenotype_database/submit-traits.html

```diff
@@ -130,14 +137,27 @@ <h5 class="modal-title" id="confirmationModalLabel">Thank you</h5>
 
         // If trait is passed as a string value, look up its fields in the phenotype table
         if (trait !== null && typeof trait === 'string') {
-            const el = document.getElementById(`trait-data-${trait}`);
-            trait = {
-                'name':              trait,
-                'dataset':           dataset,
-                'trait_name_caendr': trait,
-                'description_short': el.dataset.description,
-                'species':           el.dataset.species,
-            }
+            queryTraitByName(trait)
+                // Re-call the function using the trait value instead of the name
+                // Response check prevents infinite recursion
+                .then((response) => {
+                    if (response !== null && typeof response === 'string') throw Error(`Invalid response for trait ${id}: "${response}"`);
+                    select_trait(id, response, dataset);
+                })
+                .fail((error) => {
+                    // If an error message was received, display it, otherwise use a fallback message
+                    console.error(error);
+                    const resp = error.responseJSON
```

Comment:
if throwing from line `144`, will the property `.responseJSON` be present in the `error` object? 

Also, it might be helpful to document the expected shape of the `error.responseJSON` structure.